src/background/background-process.js

```diff
@@ -134,25 +147,20 @@ export default class BackgroundProcess {
       This way it will be guaranteed that the content script will be fully ready
       to enter the state we would like it to be.
        */
+
       await this.createTab(tabObj)
-      let tab = this.tabs.get(tabObj.uniqueId)

```

Comment:
why is this removed? We need the difference which was to setContentState (line 153)

Reply:
No it was not removed - I simply removed duplicated code - but of course I could return it

it was 
```
  async activateContent (tabObj) {
    if (!this.tabs.has(tabObj.uniqueId)) {
      /*
      This is a first activation of an extension within the tab.
      `createTab` will load a content script into the tab.
      It will take time for content script JS file and other related resources to be loaded
      and content script code to be initialized. That's why we don't want to send
      a state request (`setContentState()`) here.
      Once content script is fully activated, it will send a ContentReadyMessage.
      In this message callback we send a state message to the content script.
      This way it will be guaranteed that the content script will be fully ready
      to enter the state we would like it to be.
       */
      await this.createTab(tabObj)
      let tab = this.tabs.get(tabObj.uniqueId)
      this.setDefaultTabState(tab)
      tab.activate()
      this.updateUI(tab)
      this.notifyPageActive(tab.tabObj.tabId)
    } else {
      /*
      This is an activation on a page where content script is already loaded.
      We can send a state request to it right away.
       */
      let tab = this.tabs.get(tabObj.uniqueId)
      this.setDefaultTabState(tab)
      tab.activate()
      this.updateUI(tab)
      // Require content script to update its state
      this.setContentState(tab)
      this.notifyPageActive(tab.tabObj.tabId)
    }
  }
```

it becomes
```
  async activateContent (tabObj) {
    if (!this.tabs.has(tabObj.uniqueId)) {
      /*
      This is a first activation of an extension within the tab.
      `createTab` will load a content script into the tab.
      It will take time for content script JS file and other related resources to be loaded
      and content script code to be initialized. That's why we don't want to send
      a state request (`setContentState()`) here.
      Once content script is fully activated, it will send a ContentReadyMessage.
      In this message callback we send a state message to the content script.
      This way it will be guaranteed that the content script will be fully ready
      to enter the state we would like it to be.
       */

      await this.createTab(tabObj)
    }
    /*
    else {}
    This is an activation on a page where content script is already loaded.
    We can send a state request to it right away.
      */

    let tab = this.tabs.get(tabObj.uniqueId)
    this.setDefaultTabState(tab)
    tab.activate()
    this.updateUI(tab)
    this.notifyPageActive(tab.tabObj.tabId)
  }
```

Reply:
but in the new version, call the setContentState is missing in the scenario where we aren't creating a new tab

Reply:
oh, you are right! I am sorry! (my barin is preparing to vacations already :)) - I will return the previous version!

Reply:
Fixed